### PR TITLE
feat: add revision functionality for AM and FS

### PR DESCRIPTION
**Pull request by chrisKx0**
- add API call methods for revisions of assets, asset types, flows and flow functions
- adjust and add Mock methods and tests

### CODE REVIEW

**Comment by chrisKx0** [comment]
Fixes mache ich morgen. Brauchst es dir also noch nicht anschauen @timoklingenhoefer.

File: packages/api/lib/mock/api.mock.ts [hunk 1]
```diff
@@ -296,8 +302,10 @@ export type TimeSeriesInit = Identity<AtLeast<TimeSeries, 'id' | 'name'> & { val
 export type TaskInit = AtLeast<Task, 'name' | 'assignedTo'>;
 export type EventInit = AtLeast<Event, 'name'>;
 export type FlowInit = AtLeast<FlowDto, 'id'>;
+export type FlowRevisionInit = AtLeast<FlowRevision, 'id' | 'originalId'>;
 export type FlowDeploymentInit = AtLeast<FlowDeployment, 'id' | 'flow'>;
-export type FlowFunctionInit = AtLeast<Identity<FlowFunctionDto & { id: string }>, 'fqn' | 'id'>;
+export type FlowFunctionInit = AtLeast<Identity<FlowFunctionDto>, 'fqn'>;
```
Comment (timoklingenhoefer) [inline packages/api/lib/mock/api.mock.ts:307]: the Identity<...> is redundant here

File: .changeset/new-cups-act.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@hahnpro/hpc-api': minor
```
Comment (timoklingenhoefer) [inline .changeset/new-cups-act.md:2]: this should be major as this contains breaking changes

**Review by timoklingenhoefer — APPROVED** [review]

**Comment by 3fd** [comment]
if a release has breaking changes, please list them explicitly in the changelogs.

@chrisKx0 can you update the changelog?

**Pull request merged**